artifactory/commands/python/python.go

```diff
@@ -154,3 +169,111 @@ func (pc *PythonCommand) GetStdWriter() io.WriteCloser {
 func (pc *PythonCommand) GetErrWriter() io.WriteCloser {
 	return nil
 }
+
+func GetPypiRepoUrlWithCredentials(serverDetails *config.ServerDetails, repository string, isCurationCmd bool) (*url.URL, string, string, error) {
```

Comment:
Let's check if this function can be made private.
The same goes for the other new functions in this file.